pkg/controller/autoscaling/autoscalinggroup/setup.go

```diff
@@ -72,84 +74,90 @@ func isUpToDate(_ context.Context, obj *svcapitypes.AutoScalingGroup, obs *svcsd
 	asg := obs.AutoScalingGroups[0]
 
 	if !cmp.Equal(in.CapacityRebalance, asg.CapacityRebalance) {
-		return false, "", nil
+		return false, "spec.forProvider.capacityRebalance", nil
 	}
 	// DefaultInstanceWarmup can be updated
 	if !cmp.Equal(in.DefaultInstanceWarmup, asg.DefaultInstanceWarmup) {
-		return false, "", nil
+		return false, "spec.forProvider.defaultInstanceWarmup", nil
 	}
 	// DesiredCapacityType can be updated
 	if !cmp.Equal(in.DesiredCapacityType, asg.DesiredCapacityType) {
-		return false, "", nil
+		return false, "spec.forProvider.desiredCapacityType", nil
 	}
 	// Context is reserved
 	// if !cmp.Equal(*in.Context, *asg.Context) {
 	// 	return false, nil
 	// }
 	// DefaultCooldown can be updated
 	if !cmp.Equal(in.DefaultCooldown, asg.DefaultCooldown) {
-		return false, "", nil
+		return false, "spec.forProvider.defaultCooldown", nil
 	}
 	// DesiredCapacity can be updated
 	if !cmp.Equal(in.DesiredCapacity, asg.DesiredCapacity) {
-		return false, "", nil
+		return false, "spec.forProvider.desiredCapacity", nil
 	}
 	// HealthCheckGracePeriod can be updated
 	if !cmp.Equal(in.HealthCheckGracePeriod, asg.HealthCheckGracePeriod) {
-		return false, "", nil
+		return false, "spec.forProvider.healthCheckGracePeriod", nil
 	}
 	// HealthCheckType can be updated
 	if !cmp.Equal(in.HealthCheckType, asg.HealthCheckType) {
-		return false, "", nil
+		return false, "spec.forProvider.healthCheckType", nil
 	}
 	// MaxInstanceLifetime can be updated
 	if !cmp.Equal(in.MaxInstanceLifetime, asg.MaxInstanceLifetime) {
-		return false, "", nil
+		return false, "spec.forProvider.maxInstanceLifetime", nil
 	}
 	// MaxSize can be updated
 	if !cmp.Equal(in.MaxSize, asg.MaxSize) {
-		return false, "", nil
+		return false, "spec.forProvider.maxSize", nil
 	}
 	// MinSize can be updated
 	if !cmp.Equal(in.MinSize, asg.MinSize) {
-		return false, "", nil
+		return false, "spec.forProvider.minSize", nil
 	}
 	// NewInstancesProtectedFromScaleIn can be updated
 	if !cmp.Equal(in.NewInstancesProtectedFromScaleIn, asg.NewInstancesProtectedFromScaleIn) {
-		return false, "", nil
+		return false, "spec.forProvider.newInstancesProtectedFromScaleIn", nil
 	}
 	if !cmp.Equal(in.PlacementGroup, asg.PlacementGroup) {
-		return false, "", nil
+		return false, "spec.forProvider.placementGroup", nil
 	}
 	// VPCZoneIdentifier can be updated
 	if !cmp.Equal(in.VPCZoneIdentifier, asg.VPCZoneIdentifier) {
-		return false, "", nil
+		return false, "spec.forProvider.vpcZoneIdentifier", nil
 	}
 	// LaunchTemplate can be updated
 	if in.LaunchTemplate != nil && asg.LaunchTemplate != nil {
-		if !cmp.Equal(in.LaunchTemplate.LaunchTemplateID, asg.LaunchTemplate.LaunchTemplateId) {
-			return false, "", nil
+		if in.LaunchTemplate.LaunchTemplateID != nil {
```

Comment:
if `launchTemplateID` is not set in spec, diff will failed because `launchTemplateId` is always set in describe result.

Reply:
That's why I pull this request.

Reply:
@MisterMX PTAL again

Reply:
I think it would make more sense then to set the field in `LateInitialize` so it is reflected in spec if it is not set there. Then would not need this nil check here.